Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.h

```diff
@@ -64,7 +65,9 @@ class DeviceImpl final : public Device {
     void destroy() final;
 
     Ref<Buffer> createBuffer(const BufferDescriptor&) final;
-    Ref<Texture> createTexture(const TextureDescriptor&) final;
+    Ref<Texture> createTexture(const TextureDescriptor&, const Surface* = nullptr) final;
```

Comment:
I can move the Surface action to WebGPUSurface and related types

Reply:
I could will add `Device::createSurfaceTexture(desc, surface)` for now but should I make it a member function on Surface instead? E.g., `Surface::createTexture(desc)`? This is certainly possible, I would need to pass the device handle to the Surface since the device is the one that creates textures